### PR TITLE
Reader: Async load most streams and following management

### PR DESCRIPTION
**Pull request by blowery**
Async load most sections. To test, load each reader section, both by navigating to it using the sidebar and by loading it directly.

### CODE REVIEW

**Comment by blowery** [comment]
New sizes:

```
434423 public/build-production.9fe4a067b5816648608a.m.js.gz
 11597 public/commons.f35ddb67b5da3b3b787c.m.js.gz
90824 public/reader.b46d0063a39577ad59d1.m.js.gz
158697 public/vendor.production.m.js.gz
======
695K
```

**Comment by blowery** [comment]
Improves #11289 and targets it for landing, not master.

**Comment by samouri** [comment]
> New sizes:
what are old sizes?

File: client/reader/full-post/controller.js [hunk 1]
```diff
@@ -46,17 +46,18 @@ export function blogPost( context ) {
 	trackPageLoad( basePath, fullPageTitle, 'full_post' );
 
 	ReactDom.render(
```
Comment (samouri) [inline client/reader/full-post/controller.js]: would it be better to change this to `renderWithReduxStore` or does it not matter?

**Comment by samouri** [comment]
Functionality LGTM!

File: client/reader/full-post/controller.js (hunk 1 above)
Comment (blowery) [inline client/reader/full-post/controller.js]: yeah should switch

Comment (blowery) [inline client/reader/full-post/controller.js]: switched

**Comment by samouri** [comment]
LGTM :ship:

**Comment by mtias** [comment]
@blowery Just a general comment, but I don't think we should be async-loading sections en masse using `AsyncLoad`, which introduces a bunch of UX complications (custom placeholders), we should rely instead on the section loading system, which includes the dot-loading thing in the masterbar and the stats tracking for navigation wait time.

`AsyncLoad`, in general, should be for individual blocks or components that we consider to be heavy and of second importance to the main view of the section or the initial load of an entry point. Using `AsyncLoad` at the controller level should feel a bit weird.

I'm aware that we have issues with the flux singletons, but I hope we have a plan to go back to fix those issues so we can use regular section-loading and async-load for the inner parts of the page.

**Comment by blowery** [comment]
> I'm aware that we have issues with the flux singletons, but I hope we have a plan to go back to fix those issues so we can use regular section-loading and async-load for the inner parts of the page.

That actually turned out to be a red herring. Once a module is loaded, any duplicates of it in other chunks are ignored. 

> AsyncLoad, in general, should be for individual blocks or components that we consider to be heavy and of second importance to the main view of the section or the initial load of an entry point. Using AsyncLoad at the controller level should feel a bit weird.

I think I agree in principle, but I'm not sure I agree in practice. Until we have group-based commons, this is the best way to consolidate code that's required for all child sections. If we went back to a sections-based config, each reader chunk would increase in size dramatically, and all of that size increase would be dead code. That would slow down going from following to tags or search with no benefit.

**Pull request merged**